bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/handler/FreeAtHomeBridgeHandler.java

```diff
@@ -0,0 +1,819 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Cannot open websocket connection");
+
+            logger.debug("Cannot open websocket connection");
+
+            return;
+        }
+
+        updateStatus(ThingStatus.ONLINE);
+        FreeAtHomeBridgeHandler.freeAtHomeSystemHandler = this;
+    }
+
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters) {
+        super.handleConfigurationUpdate(configurationParameters);
+
+        // load configuration
+        FreeAtHomeBridgeHandlerConfiguration locConfig = getConfigAs(FreeAtHomeBridgeHandlerConfiguration.class);
+
+        ipAddress = locConfig.ipaddress;
+        password = locConfig.password;
+        username = locConfig.username;
+    }
+
+    /**
+     * Method to dispose
+     *
+     * @author Andras Uhrin
+     *
+     */
+    @Override
+    public void dispose() {
+    }
+
+    /**
+     * Thread that maintains connection via Websocket.
+     *
+     * @author Andras Uhrin
+     *
+     */
+    private class FreeAtHomeWebsocketMonitorThread extends Thread {
+
+        // initial delay to initiate connection
+        private AtomicInteger reconnectDelay = new AtomicInteger();
+
+        public FreeAtHomeWebsocketMonitorThread() {
+        }
+
+        @Override
+        @SuppressWarnings("null")
+        public void run() {
+            // set initial connect delay to 0
+            reconnectDelay.set(0);
+
+            try {
+                while (!isInterrupted()) {
+                    if (httpConnectionOK.get()) {
+                        if (connectSession()) {
+                            socket.awaitEndCommunication();
+
+                            logger.debug("Socket connection closed");
+
+                            reconnectDelay.set(BRIDGE_WEBSOCKET_RECONNECT_DELAY);
+                        }
+                    } else {
+                        TimeUnit.SECONDS.sleep(BRIDGE_WEBSOCKET_RECONNECT_DELAY);
+                    }
+                }
+            } catch (InterruptedException e) {
+                logger.error("Thread interrupted [{}]", e.getMessage());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Problem in websocket connection");
+            }
+        }
+
+        private boolean connectSession() throws InterruptedException {
+            int delay = reconnectDelay.get();
+
+            if (delay > 0) {
+                logger.debug("Delaying (re)connect request by {} seconds.", reconnectDelay);
+                TimeUnit.SECONDS.sleep(delay);
+            }
+
+            logger.debug("Server connecting to websocket");
+
+            socket.resetEventSocket();
+
+            if (!connectWebsocketSession()) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Problem in websocket connection");
```

Comment:
Please add I18N support.

Reply:
localisation is added to all occurrence of this statement